Source/Core/decodeGoogleEarthEnterpriseData.js

```diff
@@ -1,42 +1,38 @@
-    var keyBuffer;
-
     /**
      * Decodes data that is received from the Google Earth Enterprise server.
      *
+     * @param {ArrayBuffer} key The key used during decoding.
      * @param {ArrayBuffer} data The data to be decoded.
      *
      * @private
      */
-    function decodeGoogleEarthEnterpriseData(data) {
+    function decodeGoogleEarthEnterpriseData(key, data) {
         if (decodeGoogleEarthEnterpriseData.passThroughDataForTesting) {
             return data;
         }
 
-        if (!defined(data)) {
-            throw new DeveloperError('data is required.');
-        }
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('key', key);
+        Check.typeOf.object('data', data);
+        //>>includeEnd('debug');
 
-        var keylen = key.length;
-        if (!defined(keyBuffer)) {
-            keyBuffer = new ArrayBuffer(keylen);
-            var ui8 = new Uint8Array(keyBuffer);
-            for (var i = 0; i < keylen; ++i) {
-                ui8[i] = key.charCodeAt(i);
-            }
+        var keyLength = key.byteLength;
+        if (keyLength === 0 || (keyLength % 4) !== 0) {
+            throw new RuntimeError('The length of key must be greater than 0 and a multiple of 4.');
```

Comment:
I assume this is a `RuntimeError` because key is retrieved from whatever server we're connecting to?